WordPress/Classes/ViewRelated/QR Login/Coordinators/QRLoginCoordinator.swift

```diff
@@ -0,0 +1,67 @@
+import UIKit
+
+class QRLoginCoordinator {
+    var navigationController: UINavigationController
+
+    init(navigationController: UINavigationController = UINavigationController()) {
+        self.navigationController = navigationController
+        configureNavigationController()
+    }
+
+    func dismiss() {
+        navigationController.dismiss(animated: true)
+    }
+
+    func didScanCode(_ code: String) {
+        showVerifyAuthorization(code: code)
+    }
+
+    func showCameraScanningView(from source: UIViewController? = nil) {
+        let controller = QRLoginScanningViewController()
+        controller.coordinator = QRLoginScanningCoordinator(view: controller, parentCoordinator: self)
+
+        pushOrPresent(controller, from: source)
+    }
+
+    func showVerifyAuthorization(code: String, from source: UIViewController? = nil) {
+        print("showVerifyAuthorization:", code)
+
+        let controller = QRLoginVerifyAuthorizationViewController()
+
+        pushOrPresent(controller, from: source)
+    }
+}
+
+// MARK: - Private
+private extension QRLoginCoordinator {
+    func configureNavigationController() {
+        navigationController.isNavigationBarHidden = true
+        navigationController.modalPresentationStyle = .fullScreen
+    }
+
+    func pushOrPresent(_ controller: UIViewController, from source: UIViewController?) {
+        guard source != nil else {
```

Comment:
[nit] If we unwrap the optional here, then we can call `source.present(navigationController, animated: true)` below
```suggestion
        guard let source = source else {
```